### PR TITLE
Feature/gocart2g submodule

### PR DESCRIPTION
**Pull request by bbakernoaa**
This pulls in the NASA GOCART2G as a submodule.  It only works with the files under the `GOCART/Process_Library` but includes all of the processes needed for GOCART including the Mie scattering, wet deposition, dry deposition, emission schemes, etc.

### CODE REVIEW

**Comment by bbakernoaa** [comment]
@zmoon looks like we need to add NetCDF to the CI environment

**Review by zmoon — COMMENTED** [review]
If we can make this optional in the build, we could keep testing with `ifort` for the core stuff.

**Comment by bbakernoaa** [comment]
@zmoon please review this

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -25,16 +25,23 @@ repos:
                 src/process/old/.*\.F90|
                 src/process/Process_driver_template.F90|
                 drivers/ccpp/.*\.F90|
-                parameters/.*\.F90
+                parameters/.*\.F90|
+                src/external/GOCART/.*
```
Comment (zmoon) [inline .pre-commit-config.yaml:29]: Was this necessary? I would think the submodule should be excluded from the file search.

**Review by zmoon — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (bbakernoaa) [reply .pre-commit-config.yaml:29]: not sure but I thought it better to explicitly do this

**Pull request merged**